.github/workflows/docker-nightly-build.yaml

```diff
@@ -0,0 +1,52 @@
+name: docker-nightly-build
+
+on:
+  schedule:
+    # Set the time to be 20 mins after the pypi nightly build
+    - cron: '55 10 * * *' # 10:55am UTC, 2:55am PST, 5:55am EST
+  workflow_dispatch:
+  
+jobs:
+  
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set up Python 3.10
+        uses: actions/setup-python@v4
+      - name: Find the release version
+        run: |
+          PACKAGE_NAME="skypilot-nightly"
+          # Fetch package info from PyPI
+          PACKAGE_INFO=$(curl -s https://pypi.org/pypi/$PACKAGE_NAME/json)
+          # Parse JSON and get the latest version
+          LATEST_VERSION=$(echo $PACKAGE_INFO | jq -r '.info.version')
+
+      - name: Setup Dockerfile
+        run: |
+          touch Dockerfile
+          cat <<EOF > Dockerfile
+          FROM continuumio/miniconda3:4.11.0
+
+          # Install dependencies
+          RUN conda install -c conda-forge google-cloud-sdk && \
+              apt update -y && \
+              apt install rsync vim -y && \
+              rm -rf /var/lib/apt/lists/*
+          
+          RUN pip install "skypilot-nightly[aws,gcp,azure,kubernetes]==$LATEST_VERSION" --no-cache-dir
+          EOF
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+      - name: Login to Docker Hub
+        uses: docker/login-action@v3
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_TOKEN }}
+      - name: Build and push Docker image
+        uses: docker/build-push-action@v5
+        with:
+          context: .
+          push: true
+          tags: "${{ secrets.DOCKER_USERNAME }}/skypilot-nightly:latest,${{ secrets.DOCKER_USERNAME }}/skypilot-nightly:$LATEST_VERSION"
+          cache-from: type=gha
+          cache-to: type=gha,mode=max
```

Comment:
Nit: since running a linux/amd64 image on apple silicon can be a bit slow because of emulation (e.g., on my M2 Mac, `sky -v` takes 4 seconds on this image compared to 1 second running natively), consider adding builds for apple silicon.

IIUC from [this guide](https://docs.docker.com/build/ci/github-actions/multi-platform/), it should involve:
1. Adding the qemu action above
```
      - name: Set up QEMU
        uses: docker/setup-qemu-action@v3
```
2. In `docker/build-push-action@v5`, add:
```
platforms: linux/amd64,linux/arm64
```

Feel free to defer for a future PR.

Reply:
Good call! I just tried adding it, but it seems the `conda install -c conda-forge google-cloud-sdk` fail to be installed in the linux/arm64 version. Do you have an idea how to resolve it?